core/src/main/java/bisq/core/dao/burningman/accounting/node/full/AccountingBlockParser.java

```diff
@@ -0,0 +1,210 @@
+        // We give some extra headroom to cover potential future changes.
+        // We store value as integer in protobuf to safe space, so max. possible value is 21.47483647 BTC.
+        long maxDptValue = 6 * Coin.COIN.getValue();
+        if (inputs.size() == 1 &&
+                isValidTimeLock(tempAccountingTx, inputs.get(0)) &&
+                doAllOutputsMatchReceivers(outputs, receiverAddresses) &&
+                isWitnessDataWP2SH(inputs.get(0).getTxInWitness()) &&
+                outputs.stream().allMatch(txOutput -> isExpectedScriptType(txOutput, tempAccountingTx)) &&
+                outputs.stream().allMatch(txOutput -> txOutput.getValue() <= maxDptValue)) {
+
+            List<AccountingTxOutput> accountingTxOutputs = outputs.stream()
+                    .map(output -> new AccountingTxOutput(output.getValue(), burningManNameByAddress.get(output.getAddress())))
+                    .collect(Collectors.toList());
+            return Optional.of(new AccountingTx(AccountingTx.Type.DPT_TX, accountingTxOutputs, tempAccountingTx.getTxId()));
+        }
+
+        // BTC trade fee tx has 2 or 3 outputs.
+        // First output to receiver, second reserved for trade and an optional 3rd for change.
+        // Address check is done in parse method above already.
+        // Amounts are in a certain range but as fee can change with DAO param changes we cannot set hard limits.
+        // The min. trade fee in Nov. 2022 is 5000 sat. We use 2500 as lower bound.
+        // The largest taker fee for a 2 BTC trade is about 0.0059976 BTC (599_760 sat). We use 1_000_000 sat as upper bound.
+        // Inputs are not constrained.
+        // Max fees amount is currently 0.0176 BTC.
+        // We give some extra headroom to cover potential future fee changes.
+        // We store value as integer in protobuf to safe space, so max. possible value is 21.47483647 BTC.
+        long maxTradeFeeValue = Coin.parseCoin("0.1").getValue();
+        TempAccountingTxOutput firstOutput = outputs.get(0);
+        if (outputs.size() >= 2 &&
+                outputs.size() <= 3 &&
+                firstOutput.getValue() > 2500 &&
+                firstOutput.getValue() < 1_000_000 &&
+                isExpectedScriptType(firstOutput, tempAccountingTx) &&
+                firstOutput.getValue() <= maxTradeFeeValue) {
+            // We only keep first output.
+            String name = burningManNameByAddress.get(firstOutput.getAddress());
+            return Optional.of(new AccountingTx(AccountingTx.Type.BTC_TRADE_FEE_TX,
+                    List.of(new AccountingTxOutput(firstOutput.getValue(), name)),
+                    tempAccountingTx.getTxId()));
+        }
+
+        return Optional.empty();
+    }
+
+    // TODO not sure if other ScriptType are to be expected
+    private boolean isExpectedScriptType(TempAccountingTxOutput txOutput, TempAccountingTx accountingTx) {
```

Comment:
Why is it needed to check for script type at all?

Reply:
Its just an attempt to narrow options for false positives.